### PR TITLE
Security and Privacy Settings Re-org

### LINKED ISSUE #16727: Reorganize settings so that all privacy related settings are easily discoverable, understood and toggleable
Basically, what I have in mind is that all privacy related toggles should be next to each other in settings. More nuanced design consideration may be needed as well. We also may want one toggle that toggles all (e.g. "IP Protection Mode")

Note, at least some toggleable features with privacy implications are in the "Advanced" and not the "Security and Privacy" part of settings

### PR DESCRIPTION
**Pull request by segun**
## Explanation

Reorganize settings so that all privacy related settings are easily discoverable, understood and toggleable.

Some settings (e.g Enhanced Token Detection) are in advanced tab, they should be moved to Security and Privacy Tab
There should also be a toggle to switch off/on all security and privacy toggles.

* Fixes #16727

### Before
Screenshot 2022-12-01 at 16 05 10

### After
Screenshot 2022-12-01 at 16 05 22

## Manual Testing Steps

1. Launch MetaMask
2. Open Settings
3. Make sure all security and privacy settings (see #16727) are under the Security and Privacy Tab
4. Make sure there's a checkbox on the tab to switch off/on all checkboxes in the tab

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/settings/advanced-tab/advanced-tab.component.js [hunk 1]
```diff
@@ -665,156 +595,184 @@ export default class AdvancedTab extends PureComponent {
...
+            >
+              {t('backup')}
+            </Button>
           </div>
         </div>
       </div>
     );
   }
 
-  renderTokenDetectionToggle() {
```
Comment (segun) [inline ui/pages/settings/advanced-tab/advanced-tab.component.js:778]: The only change logic-wise in this file, is that this method (and attendant variables) was removed and moved to security tab component.

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 2]
```diff
@@ -83,6 +99,61 @@ export default class SecurityTab extends PureComponent {
     );
   }
 
+  renderIncomingTransactionsOptIn() {
```
Comment (segun) [inline ui/pages/settings/security-tab/security-tab.component.js:102]: I had to make the same arrangement for renderXXX methods here like I did in advanced tab components.

**Comment by kevinghim** [comment]
@segun please remove the `Toggle All` section

File: app/scripts/controllers/preferences.js [hunk 3]
```diff
@@ -35,6 +35,7 @@ export default class PreferencesController {
       useNonceField: false,
       usePhishDetect: true,
       dismissSeedBackUpReminder: false,
+      showGasFeeEstimationBuySwapTokens: true,
```
Comment (danjm) [inline app/scripts/controllers/preferences.js:38]: We can remove the toggle that controls this boolean from this PR

File: app/scripts/metamask-controller.js [hunk 4]
```diff
@@ -1637,6 +1637,10 @@ export default class MetamaskController extends EventEmitter {
       setUseTokenDetection: preferencesController.setUseTokenDetection.bind(
         preferencesController,
       ),
+      setShowGasFeeEstimationBuySwapTokens:
```
Comment (danjm) [inline app/scripts/metamask-controller.js:1640]: We can remove the toggle that this function controls from this PR

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 5]
```diff
@@ -167,7 +230,218 @@ export default class SecurityTab extends PureComponent {
     );
   }
 
-  renderMultiAccountBalanceCheckerOptIn() {
+  renderShowGasFeeEstimationBuySwapTokensToggle() {
```
Comment (danjm) [inline ui/pages/settings/security-tab/security-tab.component.js:233]: Let's remove this from this PR. We will add this in a future PR, when we are ready to deal with those features being toggled off.

**Review by danjm — CHANGES_REQUESTED** [review]
Code looks good. Only change needed is to remove the toggle and associated code, which I commented on

**Review by danjm — APPROVED** [review]
LGTM

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 6]
```diff
@@ -200,11 +483,44 @@ export default class SecurityTab extends PureComponent {
...
+        </div>
+        <span className="settings-page__security-tab-sub-header">
+          {this.context.t('tokenNftAutoDetection')}
+        </span>
+        <div className="settings-page__content-padded">
+          {this.renderAutoDectectTokensToggle()}
+          {this.renderBatchAccountBalanceRequestsToggle()}
+          {this.renderCollectibleDetectionToggle()}
+        </div>
         {this.renderMetaMetricsOptIn()}
```
Comment (darkwing) [inline ui/pages/settings/security-tab/security-tab.component.js:523]: I believe we're missing the title `Usage` above this line.

**Review by darkwing — CHANGES_REQUESTED** [review]

File: ui/pages/settings/security-tab/security-tab.component.js (hunk 6 above)
Comment (segun) [reply ui/pages/settings/security-tab/security-tab.component.js:523]: @darkwing if you mean the `<span className="settings-page__security-tab-sub-header">`, it was decided that we should leave this like that.

**Review by darkwing — APPROVED** [review]

**Pull request merged**